runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java

```diff
@@ -549,75 +626,98 @@ private Point findLobbyBase()
 
 	private Raid buildRaid()
 	{
-		Point gridBase = findLobbyBase();
+		Raid raid;
 
-		if (gridBase == null)
+		if (this.raid == null)
 		{
-			return null;
+			Point gridBase = findLobbyBase();
+
+			if (gridBase == null)
+			{
+				return null;
+			}
+
+			raid = new Raid(
+				new WorldPoint(client.getBaseX() + gridBase.getX(), client.getBaseY() + gridBase.getY(), LOBBY_PLANE),
+				findLobbyIndex(gridBase)
+			);
 		}
+		else
+		{
+			raid = this.raid;
+		}
+
+		// The x position of the starting room relative to other rooms
+		int baseX = raid.getLobbyIndex() % 4;
 
-		Raid raid = new Raid();
-		Tile[][] tiles;
-		int position, x, y, offsetX;
-		int startX = -2;
+		// The y position of the starting room relative to other rooms
+		int baseY = raid.getLobbyIndex() % 8 > 3 ? 1 : 0;
 
-		for (int plane = 3; plane > 1; plane--)
+		/**
+		 * i is the position of the room in the raid room array, the raid rooms are in order of north to south,
+		 * west to east, and plane 3 to plane 2. For example the 0th room is the north-west room on plane 3 and
+		 * the 15th room is the south-east room of plane 2
+		 */
+		for (int i = 0; i < raid.getRooms().length; i++)
 		{
-			tiles = client.getScene().getTiles()[plane];
+			// The x position of the current room relative to other rooms
+			// The west most room will have a value of 0 and the east most will have a value of 3
+			int x = i % 4;
+
+			// The y position of the current room relative to other rooms
+			// The north rooms will have a y value of 0 and the south will have a value of 1
+			int y = i % 8 > 3 ? 1 : 0;
+
+			// The plane of the current room
+			int plane = i > 7 ? 2 : 3;
```

Comment:
The comments are helpful, but these are still pretty magical numbers without any meaning directly associated with them.

Reply:
Let me know if the new comments make it easier to understand. I understand it very well so I am not a good judge.